README.md

```diff
@@ -249,11 +249,20 @@ This is caused by `ld` from the Conda environment shadowing the system `ld`. You
 
 **On macOS**
 
+You should use AppleClang rather than Clang here. 
+
 ```bash
 export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
 MACOSX_DEPLOYMENT_TARGET=10.9 CC=clang CXX=clang++ python setup.py develop
 ```
 
+For default macOS configuration, you can just use 
+
+```bash
+export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
+MACOSX_DEPLOYMENT_TARGET=10.9 CC=gcc CXX=g++ python setup.py develop
+```
+
```

Comment:
This is how I usually build PyTorch on MacOS for the last 3 years. I.e. if one builds using conda environment, `CMAKE_PREFIX_PATH` should already be defined. If not, no reason to tweak the variables. But I think it's important to call out that `python3` should be used, as on Intel Mac `python` still points to default Python 2.7 runtime, which is not supported for the last 3 releases of PyTorch
```suggestion
```bash
python3 setup.py develop
```

For default macOS configuration, you can just use 

```bash
export CMAKE_PREFIX_PATH=${CONDA_PREFIX:-"$(dirname $(which conda))/../"}
MACOSX_DEPLOYMENT_TARGET=10.9 CC=gcc CXX=g++ python setup.py develop
```

```

Reply:
You are right. 

Actually, the bug is in the file I'm referencing. AppleClang and Clang don't have --whole-archive option.

Reply:
`--whole-archive` is a linker option(notice, that flag is passed as `-Wl,--whole-archive`), and `-force_load` is indeed available on MacOS platform (below is the quote from `man ld`):
>      -force_load path_to_archive
>             Loads all members of the specified static archive library.  Note:
>             -all_load forces all members of all archives to be loaded.  This
>             option allows you to target a specific archive.

If you want to fix this, please do not hesitate to open a new PR and change condition from `if(CMAKE_CXX_COMPILER_ID to `if(APPLE)` or something like that

Reply:
That's true. I will open a new PR.

Reply:
New PR is here: https://github.com/pytorch/pytorch/pull/91736